pkg/agent/runner.go

```diff
@@ -830,7 +830,12 @@ func (r *Runner) DoSchedulerRequest(
 		return nil, fmt.Errorf("Bad JSON response: %w", err)
 	}
 
-	logger.Debug("Received response from scheduler", zap.Any("response", respData))
+	if respData.Permit.HasFieldLessThan(resources) {
```

Comment:
There are similar metrics added by Shayan recently: https://github.com/neondatabase/autoscaling/pull/855/files#diff-80865f7ddcbe6bce5bd007849eb00e5431d6085bb2e133f12528733b7bd8e764R54

It is reported in slightly different way - not directly through prometheus export, but through periodical "isStuck" check. 

So, we can have four outcomes:
1. Request failed
2. Request was fully denied
3. Request was partially approved
4. Request can be fully approved

Shayan's "metric" is incremented when (1) or (2) happens, yours is incremented if (2) or (3) happens.

I wonder, perhaps we actually want two metrics, one for (2), one for (3). With this distinction, the metric name could also easier convey the semantics, e. g. `autoscaling_agent_scheduler_plugin_partial_approvals_total` and `autoscaling_agent_scheduler_plugin_full_denials_total`.

Reply:
I want to clarify, what do you mean by full and partial denials? 

Do I understand correctly, that the full deny is denied on the grounds of both cpu and memory requested, and partial if on the grounds of only one type of resource?